tests/test_unstable/test_base.py

```diff
@@ -228,3 +235,98 @@ def test_raw_state_store_integration(
     new_extractor = TestExtractor(full_config, worker)
     with new_extractor:
         assert new_extractor.state_store.get_state("my-test-id") == (1, 5)
+
+
+def test_extractor_with_metrics(
+    connection_config: ConnectionConfig,
+    override_level: str | None = None,
+) -> None:
+    app_config = TestConfig(
+        parameter_one=1,
+        parameter_two="a",
+    )
+    metrics_config = MetricsConfig(
+        server=_PromServerConfig(
+            host="localhost",
+            port=9090,
+        ),
+        cognite=None,
+        push_gateways=[
+            _PushGatewayConfig(
+                host="localhost",
+                job_name="test-job",
+                username=None,
+                password=None,
+                clear_after=None,
+                push_interval=TimeIntervalConfig("30s"),
+            )
+        ],
+    )
+
+    full_config = FullConfig(
+        connection_config=connection_config,
+        application_config=app_config,
+        current_config_revision=1,
+        log_level_override=override_level,
+        metrics_config=metrics_config,
+    )
+    worker = get_checkin_worker(connection_config)
+    extractor = TestExtractor(full_config, worker, metrics=TestMetrics)
+    assert isinstance(extractor._metrics, TestMetrics) or extractor._metrics == TestMetrics
+
+    # The metrics instance should be a singleton
+    another_extractor = TestExtractor(full_config, worker, metrics=TestMetrics)
+    assert another_extractor._metrics is extractor._metrics
+    assert isinstance(another_extractor._metrics, TestMetrics) or another_extractor._metrics == TestMetrics
+
+    call_count = {"count": 0}
+    original_push = PrometheusPusher._push_to_server
+
+    def counting_push(self: "PrometheusPusher") -> None:
+        call_count["count"] += 1
+        return original_push(self)
+
+    PrometheusPusher._push_to_server = counting_push
+
+    try:
+        with extractor:
+            for pusher in extractor.metrics_push_manager.pushers:
+                assert pusher.thread is not None
+                assert pusher.thread.is_alive()
+    finally:
+        PrometheusPusher._push_to_server = original_push
+    assert call_count["count"] > 0
+
+
+def test_pushgatewayconfig_none_credentials_from_yaml(tmp_path: Path) -> None:
+    yaml_config = {
+        "push_gateways": [
+            {
+                "host": "http://localhost:9091",
+                "job_name": "test-job",
+                "clear_after": None,
+            }
+        ],
+        "cognite": None,
+        "server": None,
+    }
```

Comment:
Are you dumping a struct as YAML then reading it back? This doesn't really verify that `None` values aren't still required. Instead, you should write out the yaml and just load it directly from a string in the test.